gnc/navigator_perception/include/scan_the_code_detector.hpp

```diff
@@ -0,0 +1,126 @@
+#pragma once
+#include <string>
+#include <vector>
+#include <utility>
+#include <iostream>
+#include <algorithm>
+#include <stdexcept>
+#include <cstdint>
+#include <iterator>
+
+#include <boost/thread.hpp>
+#include <boost/bind.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+#include <opencv2/opencv.hpp>
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <Eigen/Dense>
+#include <eigen_conversions/eigen_msg.h>
+#include <Eigen/StdVector>
+
+#include <ros/ros.h>
+#include <cv_bridge/cv_bridge.h>
+#include <image_transport/image_transport.h>
+#include <image_geometry/pinhole_camera_model.h>
+#include <tf/transform_listener.h>
+#include <sensor_msgs/image_encodings.h>
+
+#include <navigator_vision_lib/cv_tools.hpp>
+#include <navigator_vision_lib/visualization.hpp>
+#include <sub8_msgs/TorpBoardPoseRequest.h>
+#include <sub8_msgs/TBDetectionSwitch.h>
+
+#include "model.h"
+#include "stereomodelfitter.h"
+
+
+// #define SEGMENTATION_DEBUG
+
+#ifdef SEGMENTATION_DEBUG
+#warning(Compiling with segmentation debugging info enabled)
+#else
+#warning(Compiling with NO segmentation debugging info)
+#endif
+
+/*
+  Warning:
+  Because of its multithreadedness, this class cannot be copy constructed. 
+  For examlple, the following will not compile:
+    ScanTheCodeDetector tb_detector = ScanTheCodeDetector();
+  Do this instead:
+    ScanTheCodeDetector tb_detector();
+*/
+
+class ScanTheCodeDetector {
+
+public:
+  ScanTheCodeDetector();
+  ~ScanTheCodeDetector();
+
+  // Public Variables
+  double image_proc_scale, feature_min_distance;
+  int diffusion_time, max_features, feature_block_size;
+
+
+private:
+
+  double model_height = .381;
```

Comment:
Can you pull these out and make that rosparams?
You can set a default value for the param if it can't find it (at least in python), so that you wouldn't have to launch from a launch file every time